File: Week5/problemA.cs
/*
Language: csharp
ID: 62414
QLink: https://quera.org/course/assignments/62414/problems
Author: @Amirkhaskar
*/

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;

public static class FareySequence
{
    public static void Main()
    {
        String input = Console.ReadLine();

        String[] spearator = { " " };
        String[] inputArray = input.Split(spearator, 2, StringSplitOptions.RemoveEmptyEntries);
        int i = 1;
        int a = 0;
        int b = 1;
        int c = 1;
        int d = int.Parse(inputArray[0]);
        if (int.Parse(inputArray[1]) == 1)
        {
            Console.WriteLine("0,1");
        }
        else
        {
            while (c <= int.Parse(inputArray[0]))
            {
                i++;
                int k = (int.Parse(inputArray[0]) + b) / d;
                int aa = a;
                int bb = b;
                a = c;
                b = d;
                c = k * c - aa;
                d = k * d - bb;
                if (int.Parse(inputArray[1]) == i)
                {
                    Console.WriteLine($"{a},{b}");
                }


            }
        }
    }

}

File: Week5/problemA.java
/*
Language: java
ID: 62414
QLink: https://quera.org/course/assignments/62414/problems
Author: @Amirkhaskar
*/

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int k = scanner.nextInt();

        int[] result = fareySequence(n, k);

        System.out.println(result[0] + "," + result[1]);
    }

    public static int[] fareySequence(int n, int k) {
        int a = 0, b = 1, c = 1, d = n;

        for (int i = 0; i < k - 1; i++) {
            int temp = (n + b) / d;
            int aa = a, bb = b;
            a = c;
            b = d;
            c = temp * c - aa;
            d = temp * d - bb;
        }

        return new int[]{a, b};
    }
}